internal/controller/policyrecommendation_registrar.go

```diff
@@ -99,13 +117,13 @@ func (r *PolicyRecommendationRegistrar) SetupWithManager(mgr ctrl.Manager) error
 		},
 	}
 
-	enqueueFunc := func(a client.Object) []reconcile.Request {
-		return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: a.GetName(), Namespace: a.GetNamespace()}}}
+	enqueueFunc := func(obj client.Object) []reconcile.Request {
+		return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: obj.GetName(),
+			Namespace: obj.GetNamespace()}}}
 	}
 
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&argov1alpha1.Rollout{}).
```

Comment:
We can retain the "For" and remove the "Watches" since "For" serves as syntactic sugar for watches.

Reply:
```// For defines the type of Object being *reconciled*, and configures the ControllerManagedBy to respond to create / delete /
// update events by *reconciling the object*.
// This is the equivalent of calling
// Watches(&source.Kind{Type: apiType}, &handler.EnqueueRequestForObject{}). ```

Reply:
Ah got it. My bad.